lib/node_modules/@stdlib/fs/append-file/docs/types/test.ts

```diff
@@ -0,0 +1,219 @@
+	appendFile( '/path/to/beepboop', 'beepboop', [], onAppend ); // $ExpectError
+	appendFile( 
+		'/path/to/beepboop',
+		'beepboop',
+		( x: number ): number => x,
+		onAppend
+	); // $ExpectError
+}
+
+// The compiler throws an error if the function is provided an `encoding` option which is not a string or null...
+{
+	appendFile( '/path/to/beepboop', 'beepboop', { encoding: 123 }, onAppend ); // $ExpectError
+	appendFile( '/path/to/beepboop', 'beepboop', { encoding: true }, onAppend ); // $ExpectError
+	appendFile( '/path/to/beepboop', 'beepboop', { encoding: false }, onAppend ); // $ExpectError
+	appendFile( '/path/to/beepboop', 'beepboop', { encoding: [] }, onAppend ); // $ExpectError
+	appendFile( '/path/to/beepboop', 'beepboop', { encoding: {} }, onAppend ); // $ExpectError
+	appendFile(
+		'/path/to/beepboop',
+		'beepboop',
+		{ encoding: ( x: number ): number => x },
+		onAppend
+	); // $ExpectError
+}
+
+// The compiler throws an error if the function is provided a `flag` option which is not a string...
+{
+	appendFile( '/path/to/beepboop', 'beepboop', { flag: 123 }, onAppend ); // $ExpectError
+	appendFile( '/path/to/beepboop', 'beepboop', { flag: true }, onAppend ); // $ExpectError
+	appendFile( '/path/to/beepboop', 'beepboop', { flag: false }, onAppend ); // $ExpectError
+	appendFile( '/path/to/beepboop', 'beepboop', { flag: null }, onAppend ); // $ExpectError
+	appendFile( '/path/to/beepboop', 'beepboop', { flag: [] }, onAppend ); // $ExpectError
+	appendFile( '/path/to/beepboop', 'beepboop', { flag: {} }, onAppend ); // $ExpectError
+	appendFile(
+		'/path/to/beepboop',
+		'beepboop',
+		{ flag: ( x: number ): number => x },
+		onAppend
+	); // $ExpectError
+}
+
+// The compiler throws an error if the function is provided a `mode` option which is not a number...
+{
+	appendFile( '/path/to/beepboop', 'beepboop', { mode: 'abc' }, onAppend ); // $ExpectError
+	appendFile( '/path/to/beepboop', 'beepboop', { mode: true }, onAppend ); // $ExpectError
+	appendFile( '/path/to/beepboop', 'beepboop', { mode: false }, onAppend ); // $ExpectError
+	appendFile( '/path/to/beepboop', 'beepboop', { mode: null }, onAppend ); // $ExpectError
+	appendFile( '/path/to/beepboop', 'beepboop', { mode: [] }, onAppend ); // $ExpectError
+	appendFile( '/path/to/beepboop', 'beepboop', { mode: {} }, onAppend ); // $ExpectError
+	appendFile(
+		'/path/to/beepboop',
+		'beepboop',
+		{ mode: ( x: number ): number => x },
+		onAppend
+	); // $ExpectError
+}
+
+// The compiler throws an error if the function is provided an unsupported number of arguments...
+{
+	appendFile(); // $ExpectError
+	appendFile( 'C:\\foo\\bar\\baz\\beepboop' ); // $ExpectError
+	appendFile( 'C:\\foo\\bar\\baz\\beepboop', 'beepboop' ); // $ExpectError
+}
+
+// Attached to main export is a `sync` method which returns an error or null...
+{
+	appendFile.sync( '/path/to/beepboop', 'beepboop' ); // $ExpectType Error | null
```

Comment:
@Planeshifter For some reason, linting on CI did not pick up this failure. The `index.d.ts` is incorrectly typed, while this assertion is correctly typed. I would have expected a CI failure.

Reply:
@kgryte Typing in `index.d.ts` matches the assertion here now, so I consider this resolved.